### ISSUE TITLE
In the haskell layer, disable ghc-mod messes important keybindings

### ISSUE THREAD

**Issue by PierreR** [syl20bnr/spacemacs#3777]
#####  How to reproduce
 * disable `ghc-mod` with : 
  ```
     (haskell :variables haskell-enable-ghc-mod-support nil)
  ```

* use `SPC-h-d-k` with `c-c c-i` (for instance)

=> it is wrongly bound to `haskell-mode-enable-process-minor-mode` 

As a result, if you fire a repl with `,sb`, you will never get the appropriate binding for `c-c c-i` (or `c-c c-t). It will always respond with "C-h f haskell-mode‘ for instruction".

##### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.0
- Spacemacs branch: develop (rev. 39c69de)
- Distribution: spacemacs
- Layers:
```elisp
(asciidoc auto-completion emacs-lisp erc dash dockerfile emacs-lisp emoji extra-langs git markdown
          (haskell :variables haskell-enable-ghc-mod-support nil)
          org
          (shell :variables shell-default-term-shell "/bin/zsh" shell-default-height 30 shell-default-position 'bottom)
          salt syntax-checking spell-checking puppet python version-control yaml)
```

**Comment by d12frosted** [comment]
When I think about it - it makes sense. We force `,sb` to run REPL. While `C-c C-l` and friends are set by `ghc-mod` or `interactive-mode` (or whatever else). So probably the best solution will be to enable `haskell-interactive-mode` when `ghc-mod` is disabled.

**Comment by PierreR** [comment]
I was about to ask "how to disable ghc-mod" in `haskell-mode` directly to https://github.com/haskell/haskell-mode/. What do you think ?

**Comment by PierreR** [comment]
I actually thought that `c-c c-t` and `c-c c-i` would be automatically rebound to `haskell-process-do-type` and `haskell-process-do-info` by `spacemacs`.

This was supposed to be fixed by #3345 but the associated commit only bind `SPC-mht` and `SPC-mhi` to mean `c-c c-t/ c-c c-i` (maybe under the assumption the `interactive-mode` would rebound these keys but it does not)

`SPC-mht` and `SPC-mhi`should be mapped (as before) to 'haskell-process-do-info and 'haskell-process-do-type' whenever an `interactive-mode` is at play.
